File: 1352-product-of-the-last-k-numbers/1352-product-of-the-last-k-numbers.java
class ProductOfNumbers {
    
    int[] nums = new int[40_001];
    int index = 0;

    public ProductOfNumbers() {
    }
    
    public void add(int num) {
        nums[index] = num;
        index++;
    }
    
    public int getProduct(int k) {
        int p = 1;
        for (int i = index - k; i < index; i++) p *= nums[i];
        return p;
    }
    
}

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers obj = new ProductOfNumbers();
 * obj.add(num);
 * int param_2 = obj.getProduct(k);
 */

File: 1352-product-of-the-last-k-numbers/1352-product-of-the-last-k-numbers.kt
class ProductOfNumbers(
    private val list:ArrayList<Int> = ArrayList()
) {

    fun add(num: Int) {
        list.add(num)
    }

    fun getProduct(k: Int): Int {
        val copyList = list.toList()
        val reverseList = copyList.reversed()

        var answer = 1
        for (index in 0 until k) {
            answer *= reverseList[index]
        }

        return answer
    }

}

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * var obj = ProductOfNumbers()
 * obj.add(num)
 * var param_2 = obj.getProduct(k)
 */